examples/create.rs

```diff
@@ -1,35 +1,97 @@
+#[macro_use]
+extern crate error_chain;
+
 extern crate network_manager;
 
 use std::env;
 use std::process;
+use std::io::Write;
+
+use network_manager::{AccessPoint, Device, DeviceType, NetworkManager};
+
+mod errors {
+    use network_manager;
+
+    error_chain! {
+        links {
+            NetworkManager(network_manager::errors::Error, network_manager::errors::ErrorKind);
+        }
+
+        errors {
+            Runtime(info: String) {
+                description("Runtime error")
+                display("{}", info)
+            }
+        }
+    }
+}
 
-use network_manager::{DeviceType, NetworkManager};
+use errors::*;
 
 fn main() {
+    if let Err(ref e) = run() {
+        let stderr = &mut ::std::io::stderr();
+        let errmsg = "Error writing to stderr";
+
+        writeln!(stderr, "{}", e).expect(errmsg);
+
+        for e in e.iter().skip(1) {
+            writeln!(stderr, "  caused by: {}", e).expect(errmsg);
+        }
+
+        ::std::process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
     let args: Vec<String> = env::args().collect();
 
-    if args.len() < 3 {
+    if args.len() != 3 {
         println!("USAGE: create SSID PASSWORD");
         process::exit(1);
     }
 
     let manager = NetworkManager::new();
 
-    let devices = manager.get_devices().unwrap();
-    let device_index = devices
-        .iter()
-        .position(|d| *d.device_type() == DeviceType::WiFi)
-        .unwrap();
-    let wifi_device = devices[device_index].as_wifi_device().unwrap();
+    let device = find_device(&manager)?;
 
-    let access_points = wifi_device.get_access_points().unwrap();
+    let wifi_device = device.as_wifi_device().unwrap();
 
-    let ap_index = access_points
+    let access_points = wifi_device.get_access_points()?;
+
+    let ap_index = find_access_point(&access_points, &args[1] as &str)?;
+
+    wifi_device.connect(&access_points[ap_index], &args[2] as &str)?;
+
+    Ok(())
+}
+
+fn find_device(manager: &NetworkManager) -> Result<Device> {
```

Comment:
I could not find a way to do that because `WiFiDevice` holds a reference to the `Device`. Again not very pretty, and if you have some ideas for how to do that better that would be excellent.